cdap-ui/app/cdap/components/Replicator/utilities/__tests__/utilities.test.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { generateTableKey, constructTablesSelection, DML } from 'components/Replicator/utilities';
+import { List, Map, Set } from 'immutable';
+
+type IColumn = Map<string, string>;
+
+describe('Replication Utilities', () => {
+  describe('Generate Table Key', () => {
+    const DB = 'dbName';
+    const TABLE = 'tableName';
+
+    const CORRECT_KEY = `db-${DB}-table-${TABLE}`;
+
+    it('should generate correct key when tableInfo is plain object', () => {
+      const tableInfo = {
+        database: DB,
+        table: TABLE,
+      };
+
+      const key = generateTableKey(tableInfo);
+      expect(key).toBe(CORRECT_KEY);
+    });
+
+    it('should  generate correct key when tableInfo is an Immutable Map', () => {
+      const tableInfo = Map({
+        database: DB,
+        table: TABLE,
+      });
+
+      const key = generateTableKey(tableInfo);
+      expect(key).toBe(CORRECT_KEY);
+    });
+  });
+
+  describe('Construct Table Selection', () => {
+    let tables: Map<string, Map<string, string>> = Map();
+    let columns: Map<string, List<IColumn>> = Map();
+    const dmlBlacklist: Map<string, Set<DML>> = Map();
+
+    beforeAll(() => {
+      // Generate tables
+      const rawTable = [
+        {
+          database: 'db1',
+          table: 'table1',
+          schema: 'schema1',
+        },
+        {
+          database: 'db1',
+          table: 'table2',
+          schema: 'schema1',
+        },
+        {
+          database: 'db1',
+          table: 'table3',
+          schema: 'schema1',
+        },
+      ];
+
+      rawTable.forEach((tableInfo) => {
+        const key = generateTableKey(tableInfo);
+        tables = tables.set(key, Map(tableInfo));
+
+        let rawColumns = List<IColumn>();
+        rawColumns = rawColumns.push(
+          Map({
+            name: `col-${tableInfo.table}-1`,
+            type: 'string',
+          })
+        );
+        rawColumns = rawColumns.push(
+          Map({
+            name: `col-${tableInfo.table}-2`,
+            type: 'string',
+          })
+        );
+
+        columns = columns.set(key, rawColumns);
+      });
+    });
+
+    it('should return empty array when no table is selected', () => {
+      const result = constructTablesSelection(null, null, null);
```

Comment:
nit: Since there are 3 arguments can there be testcases for individual validations?  For instance,
- What happens if tables is provided as an empty array?
- Can the tables be passed but with no columns?
- can `dmlBlacklist` be an empty map or object?

I am sure the type checks are handled by ts but since the tables and columns are something that we fetch from backend checking if we need additional validations

Reply:
added more tests